src/App.tsx

```diff
@@ -1,14 +1,50 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedStatus, setCompletedStatus] = useState('');
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  let visibleTodos = [...todos];
+
+  if (completedStatus === 'active') {
+    visibleTodos = todos.filter(todo => !todo.completed);
+  }
+
+  if (completedStatus === 'completed') {
+    visibleTodos = todos.filter(todo => todo.completed);
+  }
```

Comment:
this should be switch case and would be better to create separate function for all this logic